.github/workflows/pypi-wheel-builds.yml

```diff
@@ -1,36 +1,32 @@
-name: Spglib PyPI Wheel builds for linux
+name: pypi-wheel
+run-name: Spglib PyPI Wheel builds for linux
 
 on:
   push:
-    branches:
-      - rc
-      - master
+    tags: [ "v[0-9]+.[0-9]+.[0-9]+*" ]
   pull_request:
-    branches: [rc]
+    branches: [ test-PyPi-action ]
```

Comment:
Are we ok with this, or should we keep `rc` branch instead?

Reply:
I think we can remove `rc` and make `test-PyPi-action` branch, but how do you think @lan496?

Reply:
I agree with the new branch name `test-PyPi-action`. But, the current setting seems not to call `upload_pypi` job for PR to `test-PyPi-action` branch, right?

I understand the previous branch conversions were as follows:
- PR to `rc`: test to build wheels
- Push to `rc`: upload wheels to TestPyPI
- Push to `master`: upload wheels to PyPI

The current setting seems to behave as
- PR to `test-PyPi-action`: test to build wheels
- Push to `rc` with tag: upload wheels to TestPyPI
- Push to branch other than `rc` with tag: upload to PyPI

BTW, this PR becomes large now. The improvement for branching and tagging may be better to be addressed in a new PR.

Reply:
> But, the current setting seems not to call upload_pypi job for PR to test-PyPi-action branch, right?

Specifically the uploading to pypi steps, it still uploads artifacts for local testings. And yes this is intentional because at the PR stage the branch is still under development. If we truly want to make it available we push a tag.

> The current setting seems to behave as

It is much simpler:
- Any pushed tags of the form `vX.Y.Z*` or PRs to the branch name `test-PyPi-action` will trigger a build
- If it's a tag with `rc` in it (maybe can narrow the format more can't seem to regex it though) upload to TestPyPi
- If it's a tag without `rc` or `test` in it upload to PyPi

> BTW, this PR becomes large now. The improvement for branching and tagging may be better to be addressed in a new PR.

Agree, where do you want to split it off? I have another update to fix codecov almost done as well.

Reply:
@LecrisUT Thanks for your clarification. I see `github.ref` refers to the tag's name (I misread it refers to a branch name).

> Agree, where do you want to split it off? I have another update to fix codecov almost done as well.

Let's split the tags and branch parts off. The codecov fix will be proper to be included in this PR.

Reply:
> @LecrisUT Thanks for your clarification. I see github.ref refers to the tag's name (I misread it refers to a branch name).

Actually, good point, there is a variable specific for tags that will make it clearer. But I hope there is a regex way to do it instead of Contains.

> Let's split the tags and branch parts off. 

👍